tools/serverpod_cli/lib/src/util/pubspec_plus.dart

```diff
@@ -0,0 +1,90 @@
+import 'dart:io';
+import 'package:pubspec_parse/pubspec_parse.dart';
+
+import 'package:source_span/source_span.dart';
+import 'package:yaml/yaml.dart';
+
+import 'yaml_util.dart';
+
+/// A pubspec plus the yaml map it was read from, and a list of all augmented
+/// dependencies regardless of kind.
+class PubspecPlus {
+  final Pubspec pubspec;
+  final YamlMap yaml;
```

Comment:
Purpose of this class is to keep the `Pubspec` and associated yaml together. Previously a lot of methods took both a `Pubspec` and a `YamlMap` as argument, and it was implicit that the two had to be related.